dataproc/dataproc.go

```diff
@@ -0,0 +1,304 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Command dataproc uses the Cloud Dataproc API to create a cluster,
+// submit a sample job, and delete the cluster when finished.
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+
+	dataproc "google.golang.org/api/dataproc/v1"
+	storage "cloud.google.com/go/storage"
+)
+
+const (
+	// Filename of example job (Python) file uploaded to Cloud Storage
+	exampleJobName = "test-file"
+
+	// Allows for full access to Google Cloud Platform products
+	scope = "https://www.googleapis.com/auth/cloud-platform"
+
+	// Compute URI base
+	computeUriBase = "https://www.googleapis.com/compute/v1/projects/%s/zones/%s"
+)
+
+var (
+	bucketName  = flag.String("bucket", "", "GCS Bucket for storage of job file")
+	clusterName = flag.String("cluster-name", "", "Name of cluster")
+	projectID   = flag.String("project", "", "Your cloud project ID.")
+	pysparkFile = flag.String("pyspark-file", "", "PySpark Job file")
+	region      = flag.String("region", "global", "Your cloud project region.")
+	zoneID      = flag.String("zone", "", "Cloud Platform zone")
+)
+
+// clusterConfig defines the confuration of a cluster including its project, name, and region.
+type clusterConfig struct{
+		bucket,
+    project,
+		region,
+		state,
+		name,
+		uuid,
+		zone string
```

Comment:
A few of these fields are only set by the server, like uuid. I think only the user-set fields should be in a struct named `XConfig`